### PR TITLE
Update k8s HPA endpoint for k8s 1.6

### PR DESCRIPTION
**Pull request by AMeng**
Fixes https://github.com/fabric8io/kubernetes-client/issues/730

Kubernetes 1.6 changed the Horizontal Pod Autoscaler endpoint. See the [release notes](https://github.com/kubernetes/features/blob/master/release-1.6/release-notes-draft.md#autoscaling-1) for details.

### CODE REVIEW

**Comment by jstrachan** [comment]
ok to test

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/HorizontalPodAutoscalerOperationsImpl.java [hunk 1]
```diff
@@ -30,10 +30,10 @@
...
+    this(client, config, "v1", namespace, null, true, null, null, false, -1, new TreeMap<String, String>(), new TreeMap<String, String>(), new TreeMap<String, String[]>(), new TreeMap<String, String[]>(), new TreeMap<String, String>());
   }
 
   public HorizontalPodAutoscalerOperationsImpl(OkHttpClient client, Config config, String apiVersion, String namespace, String name, Boolean cascading, HorizontalPodAutoscaler item, String resourceVersion, Boolean reloadingFromServer, long gracePeriodSeconds, Map<String, String> labels, Map<String, String> labelsNot, Map<String, String[]> labelsIn, Map<String, String[]> labelsNotIn, Map<String, String> fields) {
-    super(client, config, "extensions", apiVersion, "horizontalpodautoscalers", namespace, name, cascading, item, resourceVersion, reloadingFromServer, gracePeriodSeconds, labels, labelsNot, labelsIn, labelsNotIn, fields);
+    super(client, config, "autoscaling", apiVersion, "horizontalpodautoscalers", namespace, name, cascading, item, resourceVersion, reloadingFromServer, gracePeriodSeconds, labels, labelsNot, labelsIn, labelsNotIn, fields);
```
Comment (jimmidyson) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/HorizontalPodAutoscalerOperationsImpl.java]: If this is moving to a different API group, then can you please create a new adapter and client similar to [ExtensionsAPIGroupClient](https://github.com/fabric8io/kubernetes-client/blob/master/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/ExtensionsAPIGroupClient.java) and [ExtensionsAPIGroupExtensionAdapter](https://github.com/fabric8io/kubernetes-client/blob/master/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/ExtensionsAPIGroupExtensionAdapter.java)? This keeps operations grouped similarly to the Kubernetes API groups.

**Comment by AMeng** [comment]
@jimmidyson I've created the new adapter and client, and opened a PR for the model changes here https://github.com/fabric8io/kubernetes-model/pull/205

I thought it best for clarity if I make another commit here with model updates once the other PR is merged.

**Comment by AMeng** [comment]
Seems like the new version (1.0.74) of `kubernentes-model` was not pushed up to Maven yet?

> [ERROR] Failed to execute goal on project kubernetes-client: Could not resolve dependencies for project io.fabric8:kubernetes-client:jar:2.3-SNAPSHOT: Failure to find io.fabric8:kubernetes-model:jar:1.0.74 in https://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced

**Comment by jimmidyson** [comment]
LGTM [merge]

Thanks @AMeng

**Pull request merged**